### PR TITLE
Xu's work

### PR DESCRIPTION
**Pull request by xuliugame**
add signup and login api and tests, and add password to the schema

### CODE REVIEW

File: server/src/db/tiger_bets/users.py [hunk 1]
```diff
@@ -8,4 +8,103 @@ def rebuild_user_tables():
...
+    """
+    init user table
+    """
+    if force_rebuild:
+        recreate_user_table()
+    else:
+        try:
+            get_user_by_username("test")
+        except psycopg2.errors.UndefinedTable:
+            recreate_user_table()
```
Comment (rohinivsenthil) [inline server/src/db/tiger_bets/users.py]: I don't think we need this method. `recreate_user_table` would be enough

File: server/src/db/tiger_bets/users.py [hunk 2]
```diff
@@ -8,4 +8,103 @@ def rebuild_user_tables():
...
+        self.is_deleted = is_deleted
+
+        if len(self.username) == 0 or len(self.password) == 0 or len(self.full_name) == 0 or len(self.email) == 0:
+            raise ValueError
+
+    def to_json(self):
+        return {"username": self.username, "full_name": self.full_name, "email": self.email}
+
+
+def get_user_by_username(username: str) -> int:
```
Comment (rohinivsenthil) [inline server/src/db/tiger_bets/users.py:62]: Can we fix the alignment on this file?

File: server/src/db/tiger_bets/users.py [hunk 3]
```diff
@@ -8,4 +8,103 @@ def rebuild_user_tables():
...
+def get_user_by_username_password(username: str, password: str) -> User:
+    """
+    get user by username and password
+    :param username:
+    :param password:
+    :return: user
+    """
+    user = get_user_by_username(username)
+    if user is not None and user.password == password:
+        return user
```
Comment (rohinivsenthil) [inline server/src/db/tiger_bets/users.py]: We wouldn't need a method that gets by username AND password. Just by username (get_user_by_username) is sufficient

File: server/src/db/tiger_bets/users.py (hunk 2 above)
Comment (rohinivsenthil) [inline server/src/db/tiger_bets/users.py]: The return type here should be User, not int

File: server/src/api/users.py [hunk 4]
```diff
@@ -1,8 +1,57 @@
...
+        parser = reqparse.RequestParser()
+        parser.add_argument('username', help='This field cannot be blank', required=True)
+        parser.add_argument('password', help='This field cannot be blank', required=True)
+
+        data = parser.parse_args()
+
+        user = get_user_by_username_password(data["username"], data["password"])
+        if not user:
+            return {"error": "username or password is incorrect"}, 401
+
```
Comment (rohinivsenthil) [inline server/src/api/users.py]: For our login method, we need some kind of authentication/token generation that can be added to our other APIs. See generating session key: https://www.se.rit.edu/~swen-610/00/projects/rest/chat.html

**Comment by xuliugame** [comment]
just modified

**Pull request merged**